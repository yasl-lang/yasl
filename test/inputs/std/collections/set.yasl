##set(1)\nset(0, 2, 1, 3)\nset(0, 2, 3)\nset(3)\nset(0, 2)\n2\n3\nset(1)\nset(0, 1, 3)\nset(0, 3)\nset(3)\nset(0)\n2\n2\nset(1)\nset(0, 2, 1, 3)\nset(0, 2, 3)\nset(3)\nset(0, 2)\n2\n3\nset()\nset(0, 2, 1)\nset(0, 2, 1)\nset()\nset(0, 2, 1)\n0\n3\n

const set = collections.set
let s1, let s2 = collections.set(3), collections.set(1, 2)

s1->add(1)
s2->add(2)
s2->add(0)

echo s1 & s2
echo s1 | s2
echo s1 ^ s2
echo s1 &^ s2
echo s2 &^ s1
echo len s1
echo len s2

echo s1 < s2
echo s1 > s2
echo s1 <= s2
echo s1 >= s2

let s3 = s2->copy()
s2->remove(2)

echo s1 & s2
echo s1 | s2
echo s1 ^ s2
echo s1 &^ s2
echo s2 &^ s1
echo len s1
echo len s2

echo s1 & s3
echo s1 | s3
echo s1 ^ s3
echo s1 &^ s3
echo s3 &^ s1
echo len s1
echo len s3

s1->clear()

echo s1 & s3
echo s1 | s3
echo s1 ^ s3
echo s1 &^ s3
echo s3 &^ s1
echo len s1
echo len s3

echo set(.a, .b, .c, .d, .e)->tolist()
echo set(10, 9, 8, 7, 6, 5)->tolist()

echo set(1, 2, 3) == set(1, 2, 3)
echo s3 == set(0, 2, 1)
echo s3 != set(0, 2, 1)
echo s2 != s3
echo s2 == set(0, 1)

const a = set()
const b = set(.a, .b)

echo a < b
echo a > b
echo a >= b
echo a <= b

a->add(.c)

echo a < b
echo a > b
echo a >= b
echo a <= b
