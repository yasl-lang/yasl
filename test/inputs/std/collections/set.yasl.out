set(1)
set(0, 2, 1, 3)
set(0, 2, 3)
set(3)
set(0, 2)
2
3
false
false
false
false
set(1)
set(0, 1, 3)
set(0, 3)
set(3)
set(0)
2
2
set(1)
set(0, 2, 1, 3)
set(0, 2, 3)
set(3)
set(0, 2)
2
3
set()
set(0, 2, 1)
set(0, 2, 1)
set()
set(0, 2, 1)
0
3
[a, b, c, d, e]
[10, 7, 8, 6, 9, 5]
true
true
false
true
true
true
false
false
true
false
false
false
false
